device/samples/pnp/pnpTemperatureController.js

```diff
@@ -0,0 +1,256 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+'use strict';
+
+const Protocol = require('azure-iot-device-mqtt').Mqtt;
+const Client = require('azure-iot-device').Client;
+const Message = require('azure-iot-device').Message;
+
+// String containing Hostname, Device Id & Device Key in the following formats:
+//  'HostName=<iothub_host_name>;DeviceId=<device_id>;SharedAccessKey=<device_key>'
+const deviceConnectionString = process.env.DEVICE_CONNECTION_STRING;
+
+const modelId = 'dtmi:com:example:TemperatureController;1';
+const messageSubjectProperty = '$.sub';
+const thermostat1ComponentName = 'thermostat1';
+const thermostat2ComponentName = 'thermostat2';
+const deviceInfoComponentName = 'deviceInformation';
+const commandComponentCommandNameSeparator = '*';
+let intervalToken1;
+let intervalToken2;
+let intervalToken3;
+
+const commandNameGetMaxMinReport1 = thermostat1ComponentName + commandComponentCommandNameSeparator + 'getMaxMinReport';
+const commandNameGetMaxMinReport2 = thermostat2ComponentName + commandComponentCommandNameSeparator + 'getMaxMinReport';
+const commandNameReboot = 'reboot';
+
+const commandHandler = async (request, response) => {
+  helperLogCommandRequest(request);
+  switch (request.methodName) {
+  case commandNameGetMaxMinReport1: {
+    await sendCommandResponse(request, response, 200, 'max min report from thermostat 1');
+    break;
+  }
+  case commandNameGetMaxMinReport2: {
+    await sendCommandResponse(request, response, 200, 'max min report from thermostat 2');
+    break;
+  }
+  case commandNameReboot: {
+    await sendCommandResponse(request, response, 200, 'reboot response');
+    break;
+  }
+  default:
+    await sendCommandResponse(request, response, 404, 'unknown method');
+    break;
+  }
+};
+
+const sendCommandResponse = async (request, response, status, payload) => {
+  try {
+    await response.send(status, payload);
+    console.log('Response to method \'' + request.methodName +
+              '\' sent successfully.' );
+  } catch (err) {
+    console.error('An error ocurred when sending a method response:\n' +
+              err.toString());
+  }
+};
+
+const helperLogCommandRequest = (request) => {
+  console.log('Received command request for comand name \'' + request.methodName + '\'');
+
+  if (!!(request.payload)) {
+    console.log('The command request paylaod :');
```

Comment:
> : [](start = 44, length = 2)

Your use of colon in log string is inconsistent with other uses of colon.  (Also, I think there should be a space after the colon but that's an opinion)

Reply:
removed